### PR TITLE
interesting xor solution added

### LINKED ISSUE #3051: Interesting XOR
## 🚀 Feature
(problem authored by codechef)
In this problem You are given an integer C. Let d be the smallest integer such that 2d is strictly greater than C.

Consider all pairs of non-negative integers (A,B) such that A,B<2d and A⊕B=C (⊕ denotes the bitwise XOR operation). Find the maximum value of A⋅B over all these pairs.

Input
The first line of the input contains a single integer T denoting the number of test cases. The description of T test cases follows.
The first and only line of each test case contains a single integer C.
Output
For each test case, print a single line containing one integer ― the maximum possible product A⋅B.

Constraints
1≤T≤10^5
1≤C≤10^9

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

I want to contribute under GSSoC 2021

## Pitch

I want to contribute this problems solution in  C++ language. @bhav09 sir Please assign this to me.

### PR DESCRIPTION
**Pull request by vabhv1903**
fixes #3051

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make README changes and add a new line at the EOF

**Comment by vabhv1903** [comment]
> Make README changes and add a new line at the EOF

done sir

File: C-Plus-Plus/cp/interestingxor.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+The first line of the input contains a single integer T denoting the number of test cases. The description of T test cases follows.
+The first and only line of each test case contains a single integer C.
+Output
+For each test case, print a single line containing one integer ― the maximum possible product A⋅B.
+
+Constraints
+1≤T≤105
+1≤C≤109
+
+Example Input
```
Comment (atarax665) [inline C-Plus-Plus/cp/interestingxor.cpp]: Shift the sample I/O and complexities at the end of the program.

File: C-Plus-Plus/README.md [hunk 2]
```diff
@@ -34,7 +34,9 @@
 - [Find Single Occurring element from an array](cp/find_single_occuring_element.cpp)
 - [Infix to Postfix](cp/infix_to_postfix.cpp)
 - [Intersection of Y-shaped linked list](cp/Intersection_Point_of_Y_shaped_LinkedList.cpp)
+- [interesting xor](cp/interestingxor.cpp)
 - [Interval List Intersection](cp/Interval_List_Intersection.cpp)
+
```
Comment (atarax665) [inline C-Plus-Plus/README.md:39]: Remove this extra line.

File: C-Plus-Plus/cp/interestingxor.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+{
+    int t;
+    cin >> t;
+    while (t--)
+    {
+        ll N, a, b, j;
+        string str;
+        cin >> N;
+        str = decimalToBinary(N);
+        //used to calculate length of string
```
Comment (atarax665) [inline C-Plus-Plus/cp/interestingxor.cpp]: Remove such trivial comments, use comments only where necessary.

File: C-Plus-Plus/cp/interestingxor.cpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+int main()
+{
+    int t;
+    cin >> t;
+    while (t--)
+    {
+        ll N, a, b, j;
+        string str;
+        cin >> N;
```
Comment (atarax665) [inline C-Plus-Plus/cp/interestingxor.cpp:48]: Use proper input/output statements to make your program more interactive.

File: C-Plus-Plus/cp/interestingxor.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    return 0;
+}
+
+/*
+Example Output
+70
+91
+*/
```
Comment (atarax665) [inline C-Plus-Plus/cp/interestingxor.cpp]: Add a line at EOF

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do the changes mentioned.

**Review by ankitaggarwal23 — COMMENTED** [review]
Are you still working on it.

**Comment by ankitaggarwal23** [comment]
Closing this due to inactivity since last 21 days. Thank you for your valuable time.😊

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity since last 21 days. Thank you for your valuable time.😊

**Pull request closed without merging**